### PR TITLE
feat(as-4738 remove old email field and dependencies): remove old ema…

### PR DESCRIPTION
**Pull request by Constantine-KC**
…il field and dependencies

## Ticket Number

https://pins-ds.atlassian.net/browse/AS-4738

## Description of change

Removed old email appeal field from appeal object, and updated all email dependencies so that they use the updated field appeal.email. Also, updated Contact Details by removing the email field (to be substituted by Telephone Number), and by adding the message at the top confirming the user's email address.

>> This PR is needed in order to proceed with the creation of the email page in AS-4680.

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/forms-web-app/src/validators/full-appeal/contact-details.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ function validateEmail(email) {
   throw new Error('Enter an email address in the correct format, like name@example.com');
 }
 
+// This rule was used for the email field on the page
```
Comment (edpackard-work) [inline packages/forms-web-app/src/validators/full-appeal/contact-details.js:28]: remove rule?

**Review by edpackard-work — APPROVED** [review]

File: packages/forms-web-app/src/validators/full-appeal/contact-details.js (hunk 1 above)
Comment (Constantine-KC) [reply packages/forms-web-app/src/validators/full-appeal/contact-details.js:28]: No as it is needed to validate the email coming from the new field. It is just not needed for the current PR.

Comment (Constantine-KC) [reply packages/forms-web-app/src/validators/full-appeal/contact-details.js:28]: Will move it to appropriate file once the new page has been created

File: packages/forms-web-app/__tests__/unit/validators/full-appeal/contact-details.test.js [hunk 2]
```diff
@@ -26,21 +26,21 @@ describe('validators/contact-details', () => {
       expect(rule.stack[4].options).toEqual([{ min: 2, max: 80 }]);
     });
 
-    it(`has a rule for the appellant's email`, () => {
-      const rule = rules()[1].builder.build();
+    // it(`has a rule for the appellant's email`, () => {
```
[the remark this answers is not in the document]
Comment (Constantine-KC) [reply packages/forms-web-app/__tests__/unit/validators/full-appeal/contact-details.test.js:29]: Need the tests to use them for the email validation tests once the new opage has veen developed.

**Pull request merged**